### PR TITLE
feat: Implement Managed HSM resource

### PR DESCRIPTION
**Pull request by irmatov**

### CODE REVIEW

File: resources/services/keyvault/keyvault_managed_hsm.go [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+import (
+	"context"
+
+	"github.com/cloudquery/cq-provider-azure/client"
+	"github.com/cloudquery/cq-provider-sdk/provider/schema"
+)
+
+func KeyvaultManagedHSM() *schema.Table {
+	return &schema.Table{
+		Name:         "azure_keyvault_managed_hsm",
```
Comment (roneli) [inline resources/services/keyvault/keyvault_managed_hsm.go:12]: missing PK

**Review by roneli — CHANGES_REQUESTED** [review]
missing PK

File: resources/services/keyvault/keyvault_managed_hsm.go [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+		Description:  "Managed HSM resource information with extended details.",
+		Resolver:     fetchKeyvaultManagedHSM,
+		Multiplex:    client.SubscriptionMultiplex,
+		DeleteFilter: client.DeleteSubscriptionFilter,
```
Comment (amanenk) [inline resources/services/keyvault/keyvault_managed_hsm.go:16]: Options.PrimaryKeys missing

**Review by amanenk — CHANGES_REQUESTED** [review]
LGTM. one nit

File: resources/services/keyvault/keyvault_managed_hsm.go (hunk 1 above)
Comment (irmatov) [reply resources/services/keyvault/keyvault_managed_hsm.go:12]: Fixed, thanks

File: resources/services/keyvault/keyvault_managed_hsm.go (hunk 2 above)
Comment (irmatov) [reply resources/services/keyvault/keyvault_managed_hsm.go:16]: Fixed, thanks

**Review by amanenk — APPROVED** [review]

**Review by roneli — APPROVED** [review]

**Pull request merged**